File: summer2018/easy/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(!(x%10) && x > 1) return false;
        int tmp = 0;
        while(x > tmp){
            tmp = tmp*10 + x%10;
            x  /= 10;
        }
        return (x == tmp) || (x == tmp/10);
    }
};

File: language/python/9-palindrome-number.py
# My method (152ms)
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        tmp = x
        num = 0
        while x > 0:
            num = num * 10 + x % 10
            x /= 10
        return num == tmp
# Other method (fastest 136ms) 
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        string = ""
        string += str(x)

        if string == string[::-1]:
            return True
        else:
            return False
# Super short code
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        return str(x)[::-1] == str(x)
